main.tf

```diff
@@ -100,51 +79,61 @@ resource "aws_iam_policy" "iam_policy" {
 
   policy = jsonencode({
     Version = "2012-10-17",
-    Statement = concat(
-      var.folder_monitoring_enabled ? [
-        {
-          Effect = "Allow",
-          Action = [
-            "s3:ListBucket",
-            "s3:PutObject"
-          ],
-          Resource = [
-            "arn:aws:s3:::${var.snowflake_aws_s3_stage_bucket_name}",
-            "arn:aws:s3:::${var.snowflake_aws_s3_stage_bucket_name}/*"
-          ]
-        }
-      ] : [],
-      [
-        {
-          Effect = "Allow",
-          Action = [
-            "sqs:DeleteMessage",
-            "sqs:GetQueueUrl",
-            "sqs:ListQueues",
-            "sqs:ChangeMessageVisibility",
-            "sqs:SendMessageBatch",
-            "sqs:ReceiveMessage",
-            "sqs:SendMessage",
-            "sqs:DeleteMessageBatch",
-            "sqs:ChangeMessageVisibilityBatch"
-          ],
-          Resource = [
-            "arn:aws:sqs:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:${var.sqs_queue_name}"
-          ]
-        },
-        {
-          Effect = "Allow",
-          Action = [
-            "logs:PutLogEvents",
-            "logs:CreateLogStream",
-            "logs:DescribeLogStreams"
-          ],
-          Resource = [
-            "arn:aws:logs:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:log-group:${local.cloudwatch_log_group_name}:*"
-          ]
-        }
-      ]
-    )
+    Statement = [
+      {
+        Effect = "Allow",
+        Action = [
+          "s3:ListBucket",
+          "s3:PutObject",
+          "s3:GetObject"
+        ],
+        Resource = [
+          "arn:aws:s3:::${var.snowflake_aws_s3_bucket_name}/",
+          "arn:aws:s3:::${var.snowflake_aws_s3_bucket_name}/*"
+        ]
+      },
+      {
+        Effect = "Allow",
+        Action = ["s3:GetObject"],
+        Resource = [
+          "arn:aws:s3:::${var.snowflake_aws_s3_bucket_name}/*/shredding_complete.json"
+        ]
+      },
+      {
+        Effect = "Allow",
+        Action = [
+          "sqs:DeleteMessage",
+          "sqs:GetQueueUrl",
+          "sqs:ListQueues",
+          "sqs:ChangeMessageVisibility",
+          "sqs:ReceiveMessage",
+          "sqs:SendMessage"
```

Comment:
I wonder why `sqs:SendMessage` is in this list.  I cannot think why the loader needs this.

Reply:
Good point!  If the code base never invokes this should be safe to remove.  Equally feels like a fairly safe one to leave in?